src/lib/battery/battery.cpp

```diff
@@ -231,45 +234,42 @@ void Battery::estimateStateOfCharge(const float voltage_v, const float current_a
 				{3.69, 5.9},
 				{3.50, 0.0}
 			};
-			bool above_min = true;
-			bool below_max = true;
-
-
 
 			if (cell_voltage_filtered >= SOCLookup[0].OCVoltage) {
-				_state_of_charge = 1.0;
-				below_max = false;
+				soc_initial = 1.0;
 			}
-			if (cell_voltage_filtered <= SOCLookup[10].OCVoltage) {
-				_state_of_charge = 0.0;
-				above_min = false;
+			else if (cell_voltage_filtered <= SOCLookup[lookup_size].OCVoltage) {
+				soc_initial = 0.0;
 			}
-			if (above_min && below_max) {
-				for (int i = 0; i < 10; i++) {
+			else {
+				for (int i = 1; i < lookup_size; i++) {
 					float voltage_key = SOCLookup[i].OCVoltage;
 					if (cell_voltage_filtered > voltage_key) {
 						const float volt1 = SOCLookup[i].OCVoltage;
 						const float soc1 = SOCLookup[i].SOC;
-						const float volt2 = SOCLookup[i+1].OCVoltage;
-						const float soc2 = SOCLookup[i+1].SOC;
-						soc_initial = (soc1 + (soc2 - soc1) * (cell_voltage_filtered - volt1) / (volt2 - volt1))/100;
+						const float volt2 = SOCLookup[i-1].OCVoltage;
+						const float soc2 = SOCLookup[i-1].SOC;
+						soc_initial = math::gradual(cell_voltage_filtered, volt1, volt2, soc1, soc2)/100;
```

Comment:
Fixed bug, as the lookup table is reversed (max is 0th element, min is 11th), i+1 should be i-1 in the lookup search to select the values either side of the queried voltage.